consensus/polybft/consensus_runtime.go

```diff
@@ -313,7 +330,11 @@ func (c *consensusRuntime) populateFsmIfBridgeEnabled(
 func (c *consensusRuntime) FSM() error {
 	// figure out the parent. At this point this peer has done its best to sync up
 	// to the head of their remote peers.
-	parent, epoch := c.getLastBuiltBlockAndEpoch()
+	c.lock.RLock()
+	proposerSnapshot := c.proposerSnapshot.Copy()
+	parent, epoch := new(types.Header), new(epochMetadata)
+	*parent, *epoch = *c.lastBuiltBlock, *c.epoch
```

Comment:
It's not a safe copy for c.lastBuiltBlock. Could you use c.lastBuiltBlock.Copy?

Reply:
For epoch also

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1018/commits/4d4fff05b2f2df8263b3a8903d32fe77e2853e17